src/Attributes/src/Internal/Instantiator/Factory.php

```diff
@@ -55,6 +63,10 @@ public function instantiate(\ReflectionClass $attr, array $arguments, string $co
      */
     private function isNamedArguments(\ReflectionClass $class): bool
     {
-        return \is_subclass_of($class->getName(), NamedArgumentConstructorAnnotation::class);
+        return is_subclass_of($class->getName(), NamedArgumentConstructorAnnotation::class)
+            || (
+                class_exists(NamedArgumentConstructor::class)
+                && null !== $this->annotationReader->getClassAnnotation($class, NamedArgumentConstructor::class)
```

Comment:
Instantiation by annotation cannot be done on attributes.

Moreover, this annotation is not an attribute, so it cannot be read by an attribute reader.

That is, the case is not taken into account:
```php
#[NamedArgumentConstructor]
#[\Attribute(\Attribute::TARGET_ALL)]
class MyAttribute
{
    public function __construct(string $some) { ... }
}
```